### PR TITLE
Add filtering and serach for Registry Entries

### PR DESCRIPTION
**Pull request by vithu30**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/EndpointRegistry.java [hunk 1]
```diff
@@ -81,8 +81,10 @@ EndpointRegistryInfo getEndpointRegistryByUUID(String registryId, String tenantD
      * @return A list of EndpointRegistryEntry objects
      * @throws APIManagementException if failed to get entries of an Endpoint Registry
      */
-    List<EndpointRegistryEntry> getEndpointRegistryEntries(String sortBy, String sortOrder, int limit, int offset,
-                                                           String registryId) throws APIManagementException;
+    List<EndpointRegistryEntry> getEndpointRegistryEntries(String sortBy, String sortOrder, int limit,
```
Comment (fazlan-nazeem) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/EndpointRegistry.java:84]: lets change javadoc accordingly

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/EndpointRegistryImpl.java [hunk 2]
```diff
@@ -147,9 +147,12 @@ public EndpointRegistryEntry getEndpointRegistryEntryByUUID(String registryId, S
      * @throws APIManagementException if failed to get entries of an Endpoint Registry
      */
     public List<EndpointRegistryEntry> getEndpointRegistryEntries(String sortBy, String sortOrder, int limit,
-                                                                  int offset, String registryId)
+                                                                  int offset, String registryId, String serviceType,
```
Comment (fazlan-nazeem) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/EndpointRegistryImpl.java:150]: javadoc

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/EndpointRegistry.java (hunk 1 above)
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/EndpointRegistry.java:88]: Fixed in 52c8640

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/EndpointRegistryImpl.java (hunk 2 above)
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/EndpointRegistryImpl.java:154]: Fixed in 52c8640

**Pull request merged**